File: TGAPEngine/Scene.cpp
#include "Scene.h"

namespace engine{
	Scene::Scene()
	{
	}


	Scene::~Scene()
	{
	}
	void Scene::removeEntity(Entity *e){
		std::vector<Entity *>::iterator it = entities.begin();
		while (it != entities.end()){
			if ((*it) == e){
				//Entity * removedEntity = ()
				it = entities.erase(it);
				break;
			}
			else{
				++it;
			}

		}
	}
	void Scene::addEntity(Entity *e)
	{
		entities.emplace_back(e);
	}
	void Scene::draw(DrawingContext * context){
		for (std::vector<Entity *>::iterator it = entities.begin(); it != entities.end(); ++it)

		{
			(*it)->draw(glm::mat4(1.0f),context);
		}
	}
	Entity * Scene::findEntityByName(std::string name){
		for (std::vector<Entity *>::iterator it = entities.begin(); it != entities.end(); ++it)

		{
			if ((*it)->name == name){
				return (*it);
			}
		}
		return nullptr;
	}
}

File: CS/TGAPE_CS/Engine/Game/Scene.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;
using System.Text;
using System.Threading.Tasks;
using TGAPE_CS.Engine.Renderer;

namespace TGAPE_CS.Engine.Game
{
    public class Scene
    {
        public List<Entity> Entities { get; set; }
        public Scene()
        {
            Entities = new List<Entity>();
        }
        public Scene(IEnumerable<Entity> entities)
        {
            Entities = new List<Entity>(entities);
        }

        public void Draw(DrawingContext context)
        {
            foreach(var entity in Entities)
            {
                entity.Draw(Matrix4x4.Identity, context);
            }
        }
        public Entity FindEntityByName(string name)
        {
            return Entities.FirstOrDefault(x => x.name.Equals(name));
        }
    }
}
